src/main/java/org/opensearch/flowframework/workflow/CreateIngestPipelineStep.java

```diff
@@ -0,0 +1,178 @@
+                    logger.error("Failed to create ingest pipeline : " + e.getMessage());
+                    createIngestPipelineFuture.completeExceptionally(e);
+                }
+                break;
+            }
+        }
+
+        // Create PutPipelineRequest and execute
+        if (configuration != null) {
+            PutPipelineRequest putPipelineRequest = new PutPipelineRequest(pipelineId, configuration, XContentType.JSON);
+            clusterAdminClient.putPipeline(putPipelineRequest, ActionListener.wrap(response -> {
+                logger.info("Created ingest pipeline : " + putPipelineRequest.getId());
+
+                // PutPipelineRequest returns only an AcknowledgeResponse, returning pipelineId instead
+                createIngestPipelineFuture.complete(new WorkflowData() {
+                    @Override
+                    public Map<String, Object> getContent() {
+                        return Map.of("pipelineId", putPipelineRequest.getId());
+                    }
+                });
+
+                // TODO : Use node client to index response data to global context (pending global context index implementation)
+
+            }, exception -> {
+                logger.error("Failed to create ignest pipeline : " + exception.getMessage());
+                createIngestPipelineFuture.completeExceptionally(exception);
+            }));
+        } else {
+            // Required workflow data not found
+            createIngestPipelineFuture.completeExceptionally(new Exception("Failed to create ingest pipeline, required inputs not found"));
+        }
+
+        return createIngestPipelineFuture;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    /**
+     * Temporary, generates the ingest pipeline request content from workflow data
+     * {
+     *  "description" : "<description>",
+     *  "processors" : [
+     *      "text_embedding" : {
+     *          "model_id" : "<model_id>",
+     *          "field_map" : {
+     *              "<input_field_name>" : "<output_field_name>"
+     *          }
+     *      }
+     *  ]
+     * }
+     *
+     * @param description The description of the ingest pipeline configuration
+     * @param modelId The ID of the model that will be used in the embedding interface
+     * @param inputFieldName The field name used to cache text for text embeddings
+     * @param outputFieldName The field name in which output text is stored
+     * @throws IOException if the request content fails to be generated
+     * @return the xcontent builder with the formatted ingest pipeline configuration
+     */
+    private XContentBuilder buildIngestPipelineRequestContent(
+        String description,
+        String modelId,
+        String inputFieldName,
+        String outputFieldName
+    ) throws IOException {
+        return XContentFactory.jsonBuilder()
+            .startObject()
+            .field("description", description)
```

Comment:
Let's make all of these constants like `DESCRIPTION_FIELD` since you'll need a corresponding `parse()` method relying on them.  See my template parsing methods in #47 for examples.